### PR TITLE
Use focus-visible instead of focus pseudo selector

### PR DESCRIPTION
**Pull request by vovakulikov**
Resolves part of https://github.com/sourcegraph/sourcegraph/issues/44878

## Context

- `:focus` is unconditional selector, every time when element is focused (regardless of what caused this focus) this selector will match
- `:focus-visible` is conditional, and browser decides when this selector will match, for example clicks on button shouldn't trigger focus-ring style, so focus-visible won't be triggered/matched.

In our styles we're heavily using `:focus` selector, and therefore we have a lot of controls which always have focus-ring styles, sometimes this also causes having double focus-ring UI (our custom ring from `base.scss` and browser standard ring)

So this PR replaces `:focus` to `:focus-visible` in some places where it makes sense, we still need to use :focus somewhere where it should be use (like search boxes, input, ..etc)

### Difference between `:focus-visible` and `.focus-visible`
Different browsers have different heuristics about showing focus ring, for example select UI does have focus ring if you click it in Chrome, but it doesn't have focus ring in Safari. So to unify this behaviour and support browsers in which :focus-visible is still unavailable (very small percentage but still) we use focus-visible polyfill and focus-visible postcss plugin.

In general this works in a way that we track all clicks and keyboards strokes and add `.focus-visible` class to focusable elements (only if we navigate to them through keyboard)

## Test plan
- Check that clicking on buttons doesn't trigger focus ring appearance 
- Check that clicking on select doesn't trigger focus ring appearance 
- Check that both buttons and select controls still have focus ring if you're navigation though keyboard

### CODE REVIEW

**Review by eseliger — APPROVED** [review]

File: BUILD.bazel [hunk 1]
```diff
@@ -70,7 +70,6 @@ js_library(
     deps = [
         "//:node_modules/autoprefixer",
         "//:node_modules/postcss-custom-media",
-        "//:node_modules/postcss-focus-visible",
```
Comment (sqs) [inline BUILD.bazel:73]: Did you mean to also remove postcss-focus-visible from postcss.config.js?

Comment (sqs) [reply BUILD.bazel:73]: Or actually after reading the PR description, I think this bazel change in this PR should be reverted.

Comment (sqs) [reply BUILD.bazel:73]: https://github.com/sourcegraph/sourcegraph/pull/57439

Comment (sqs) [reply BUILD.bazel:73]: (I was getting Bazel errors on CI. Not sure why it didn't happen here.)

**Pull request merged**